core/core.go

```diff
@@ -211,8 +212,17 @@ func (n *IpfsNode) startOnlineServices(ctx context.Context, routingOption Routin
 		}()
 	}
 
+	addrsFactory, err := makeAddrsFactory(cfg.Addresses)
+	if err != nil {
+		return err
+	}
+
+	hostopts := &ConstructPeerHostOpts{
+		AddrsFactory:      addrsFactory,
+		DisableNatPortMap: cfg.Swarm.DisableNatPortMap,
+	}
 	peerhost, err := hostOption(ctx, n.Identity, n.Peerstore, n.Reporter,
```

Comment:
Once libp2p/go-libp2p#197 is merged I'd like to remove ConstructPeerHostOpts and use basichost.HostOpts, and would like to do the same with regard to swarm.